ui/app/pages/swaps/swaps.util.js

```diff
@@ -35,13 +35,18 @@ const getBaseApi = function (type) {
 
 const validHex = (string) => Boolean(string?.match(/^0x[a-f0-9]+$/u))
 const truthyString = (string) => Boolean(string?.length)
+const stringIsValidDecimalNumber = (string) => !isNaN(string) && string.match(/^[.0-9]+$/u) && !isNaN(parseFloat(string))
```

Comment:
`string.match(/^[.0-9]+$/u)` allows more than one "."; also, do we need both `!isNan` checks?  Doesn't one identify a problem?

Also, I think we should try to make these functions that we can write tests for.